store/store.go

```diff
@@ -835,30 +879,22 @@ func (s *Store) Download(name string, downloadInfo *snap.DownloadInfo, pbar prog
 		url = downloadInfo.DownloadURL
 	}
 
-	req, err := s.newRequest("GET", url, nil, user)
-	if err != nil {
-		return "", err
-	}
-
-	if err := download(name, w, req, pbar); err != nil {
+	if err := download(name, url, user, s, w, pbar); err != nil {
 		return "", err
 	}
 
 	return w.Name(), w.Sync()
 }
 
 // download writes an http.Request showing a progress.Meter
-var download = func(name string, w io.Writer, req *http.Request, pbar progress.Meter) error {
+var download = func(name, url string, user *auth.UserState, s *Store, w io.Writer, pbar progress.Meter) error {
 	client := &http.Client{}
 
-	resp, err := client.Do(req)
-	if err != nil {
```

Comment:
Why was this logic dropped?

The line below, defer-closing the Body, will explode if the Body is nil.